prefect_kubernetes/worker.py

```diff
@@ -0,0 +1,671 @@
+
+    def _slugify_labels(self):
+        """Slugifies the labels in the job manifest."""
+        all_labels = {**self.job_manifest["metadata"].get("labels", {}), **self.labels}
+        self.job_manifest["metadata"]["labels"] = {
+            _slugify_label_key(k): _slugify_label_value(v)
+            for k, v in all_labels.items()
+        }
+
+    def _populate_image_if_not_present(self):
+        """Ensures that the image is present in the job manifest. Populates the image
+        with the default Prefect image if it is not present."""
+        try:
+            if (
+                "image"
+                not in self.job_manifest["spec"]["template"]["spec"]["containers"][0]
+            ):
+                self.job_manifest["spec"]["template"]["spec"]["containers"][0][
+                    "image"
+                ] = get_prefect_image_name()
+        except KeyError:
+            raise ValueError(
+                "Unable to verify image due to invalid job manifest template."
+            )
+
+    def _populate_command_if_not_present(self):
+        """
+        Ensures that the command is present in the job manifest. Populates the command
+        with the `prefect -m prefect.engine` if a command is not present.
+        """
+        try:
+            command = self.job_manifest["spec"]["template"]["spec"]["containers"][
+                0
+            ].get("args")
+            if command is None:
+                self.job_manifest["spec"]["template"]["spec"]["containers"][0][
+                    "args"
+                ] = [
+                    "python",
+                    "-m",
+                    "prefect.engine",
+                ]
+            elif isinstance(command, str):
+                self.job_manifest["spec"]["template"]["spec"]["containers"][0][
+                    "args"
+                ] = command.split()
+            elif not isinstance(command, list):
+                raise ValueError(
+                    "Invalid job manifest template: 'command' must be a string or list."
+                )
+        except KeyError:
+            raise ValueError(
+                "Unable to verify command due to invalid job manifest template."
+            )
+
+    def _populate_generate_name_if_not_present(self):
+        """Ensures that the generateName is present in the job manifest."""
+        manifest_generate_name = self.job_manifest["metadata"].get("generateName", "")
+        has_placeholder = len(find_placeholders(manifest_generate_name)) > 0
+        if not manifest_generate_name or has_placeholder:
+            generate_name = None
+            if self.name:
+                generate_name = _slugify_name(self.name)
+            # _slugify_name will return None if the slugified name in an exception
+            if not generate_name:
+                generate_name = "prefect-job"
+            self.job_manifest["metadata"]["generateName"] = f"{generate_name}-"
+
+
+class KubernetesWorkerVariables(BaseVariables):
```

Comment:
are these (will these be) capital `V` Prefect `Variables`?  or is that class just to make it easy to set the most important configuration on jobs created by this worker

Reply:
These will be the attributes that deployment creators can specify in `infra_overrides`. In the future, users will be able to provide a Prefect Variable to use as the value for one of these attributes, but these variables won't be useable in other contexts.

This class doesn't get instantiated anywhere. It's used to generate an OpenAPI schema that we can use for validation and generating a form in the UI.